torch/csrc/autograd/python_variable.cpp

```diff
@@ -809,8 +809,70 @@ static PyMethodDef extra_methods[] = {
   {nullptr}
 };
 
+/* From https://github.com/python/cpython/blob/v3.7.0/Modules/xxsubtype.c
+   If compiled as a shared library instead, some compilers don't allow addresses
+   of Python objects defined in other libraries to be used in static
+   initializers here.  The DEFERRED_ADDRESS macro is used to tag the slots where
+   such addresses appear; the module init function must fill in the tagged slots
+   at runtime.  The argument is for documentation -- the macro ignores it.
+*/
+#define DEFERRED_ADDRESS(ADDR) nullptr
+
+struct THPVariableMeta {
+  PyHeapTypeObject base;
+};
+
+int THPVariableMetaType_init(PyObject *cls, PyObject *args, PyObject *kwargs) {
+  if (PyType_Type.tp_init(cls, args, kwargs) < 0) {
+    return -1;
+  }
+  // TODO: put something nontrivial here
+  return 0;
+}
+
+PyTypeObject THPVariableMetaType = {
+  PyVarObject_HEAD_INIT(DEFERRED_ADDRESS(&PyType_Type), 0)
```

Comment:
Is this `PyVarObject_HEAD_INIT` doing anything non-trivial? Because it would do it before the `PyType_Type` is actually populated during the init here no?

Actually that field is not updated in the init. So it shouldn't be a DEFERRED_ADDRESS ?

Reply:
It gets automatically initialized by `tp_base`. https://github.com/python/cpython/blob/v3.7.0/Modules/xxsubtype.c#L263-L266